tune_sklearn/tune_basesearch.py

```diff
@@ -232,16 +236,19 @@ def __init__(self,
                  error_score="raise",
                  return_train_score=False,
                  local_dir="~/ray_results",
-                 max_iters=None,
+                 max_iters=1,
                  use_gpu=False):
-
+        if max_iters < 1:
+            raise ValueError("max_iters must be greater than or equal to 1.")
         self.estimator = estimator
 
-        if not self._can_early_stop() and max_iters is not None:
```

Comment:
I think there's actually more cases we need to be checking here. Here's all the cases:

- User wants to early stop, and it can be done
  - **need to make sure the user sets `max_iters`, so raise a warning if this isn't done**
- User wants to early stop, and it cannot be done
  - directly throw an error, regardless of what `max_iters` is
  - **maybe it would be nice the remove the warning that would come up in the `if not self._can_early_stop() and max_iters > 1:` since there will be duplicate errors**
- User does not want to early stop, and it can be done
  - **if `max_iters` isn't set, do nothing**
  - **if `max_iters` is set, raise a warning that it is ignored because user didn't enable early stop**
- User does not want to early stop, and it cannot be done
  - **if `max_iters` isn't set, do nothing**
  - **if `max_iters` is set, raise a warning that it is ignored because user didn't enable early stop**

Right now

Reply:
It is now reduced to this:

- User wants to early stop, and it cannot be done
  - directly throw an error.

- User wants to early stop, and it can be done, and max_iters is not set
  - raise a warning that it should be set.

- regardless of whether it can be done, if user does not want to early stop and if `max_iters` is set: 
  - raise a warning that it is ignored / set to 1.